### PR TITLE
Align actuals timeseries with the projections timeseries.

### PR DESCRIPTION
**Pull request by mikelehen**

### CODE REVIEW

File: src/models/Projection.ts [hunk 1]
```diff
@@ -166,12 +172,12 @@ export class Projection {
...
       const latestCurrentUssage = this.lastValue(
-        this.actualTimeseries.map(row => row.ICUBeds.currentUsageTotal),
+        this.actualTimeseries.map(row => row && row.ICUBeds.currentUsageTotal),
       );
       const latestUsageCovid = this.lastValue(
-        this.actualTimeseries.map(row => row.ICUBeds.currentUsageCovid),
+        this.actualTimeseries.map(row => row && row.ICUBeds.currentUsageCovid),
       );
-      return latestCurrentUssage - latestUsageCovid;
+      return latestCurrentUssage! - latestUsageCovid!;
```
Comment (goldblatt) [inline src/models/Projection.ts:180]: lol

File: src/models/Projection.ts [hunk 2]
```diff
@@ -229,12 +235,29 @@ export class Projection {
...
+  private getActualTimeseries(
+    actuals: ActualsTimeseries,
+  ): Array<CANActualsTimeseriesRow | null> {
+    if (actuals.length === 0) {
+      return actuals;
     }
+
+    const firstDate = moment(this.dates[0]);
+    const firstActualDate = moment(actuals[0].date);
+    const padFrontDays = firstActualDate.diff(firstDate, 'days');
```
Comment (goldblatt) [inline src/models/Projection.ts:247]: is the data daily for sure now?

Comment (goldblatt) [reply src/models/Projection.ts:247]: otherwise the dates might not line up is what i'm thinking jst seems tenuous

**Comment by mikelehen** [comment]
I reworked this completely to basically walk the two timeseries in parallel and match trim / pad / fill in missing dates in the actuals.  I added a bunch of asserts and verified that the /all page doesn't crash, so I think it probably is working as intended. :-)

But I don't like this approach, so feel free to come up with something better.

**Pull request closed without merging**